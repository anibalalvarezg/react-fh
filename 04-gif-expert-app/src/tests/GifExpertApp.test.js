import React from 'react'
import { shallow } from 'enzyme';
import { GifExpertApp } from '../GifExpertApp';
import '@testing-library/jest-dom';

describe('Test for GifExpertApp', () => {
    test('should  display component correctly', () => {
        const wrapper = shallow(<GifExpertApp />);
        expect( wrapper ).toMatchSnapshot();
    });
    
    test('should display a categories list', () => {
        const categories = ['Rick and Morty', 'Dragon Ball']; 
        const wrapper = shallow( <GifExpertApp defaultCategories={ categories } />);
        expect( wrapper ).toMatchSnapshot();
        expect( wrapper.find('GifGrid').length ).toBe( categories.length )
    });
});