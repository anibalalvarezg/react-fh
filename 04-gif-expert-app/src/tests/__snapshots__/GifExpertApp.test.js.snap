// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test for GifExpertApp should  display component correctly 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;

exports[`Test for GifExpertApp should display a categories list 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="Rick and Morty"
      key="Rick and Morty"
    />
    <GifGrid
      category="Dragon Ball"
      key="Dragon Ball"
    />
  </ol>
</Fragment>
`;
